tests/unit/test_messages_emitter.py

```diff
@@ -181,13 +202,12 @@ def test_init_developer_modes(mode, tmp_path, monkeypatch):
     assert handler.mode == mode
 
 
-@pytest.mark.parametrize("method_name", [x for x in dir(Emitter) if x[0] != "_" and x != "init"])
-def test_needs_init(method_name):
+def test_needs_init(emitter_methods):
```

Comment:
A problem with this approach is that now the test is no longer parametrized; so instead of having N tests we have a single test that looks but will stop as soon as an assertion fails.

How about, instead of having a fixture, just a regular function that's called during `parametrize`? Like:
```python
@pytest.mark.parametrize("method_name", emitter_methods(exclude=["init"]))
```